packages/basic-crawler/src/internals/basic-crawler.ts

```diff
@@ -989,18 +1000,33 @@ export class BasicCrawler<Context extends CrawlingContext = BasicCrawlingContext
         }
     }
 
-    protected async _handleFailedRequestHandler(crawlingContext: Context, error: Error): Promise<void> {
-        if (this.failedRequestHandler) {
-            await this._tagUserHandlerError(() => this.failedRequestHandler?.({ ...crawlingContext, error }, error));
-            return;
-        }
-
+    protected async _handleFailedRequestHandler(crawlingContext: Context, error: unknown): Promise<void> {
+        // Always log the last error regardless if the user provided a failedRequestHandler
         const { id, url, method, uniqueKey } = crawlingContext.request;
-        const message = error instanceof TimeoutError && !process.env.CRAWLEE_VERBOSE_LOG ? error.message : error.stack;
+        const message = error instanceof TimeoutError && !process.env.CRAWLEE_VERBOSE_LOG
+            ? error.message : (error as any).stack ?? (error as any).message ?? error;
```

Comment:
I really don't like those `??` inside the ternary. Especially with the TS overrides. Can we make this a bit more readable with more variables?